pkg/analysis/passes/backenddebug/backenddebug_test.go

```diff
@@ -0,0 +1,96 @@
+package backenddebug
+
+import (
+	"github.com/grafana/plugin-validator/pkg/analysis"
+	"github.com/grafana/plugin-validator/pkg/analysis/passes/archive"
+	"github.com/grafana/plugin-validator/pkg/analysis/passes/metadata"
+	"github.com/grafana/plugin-validator/pkg/testpassinterceptor"
+	"github.com/stretchr/testify/require"
+	"path/filepath"
+	"strings"
+	"testing"
+)
+
+func TestBackendDebug(t *testing.T) {
+	var (
+		pluginJSONWithExecutable = []byte(`{"executable": "gpx_plugin"}`)
+		pluginJSONEmpty          = []byte(`{}`)
+	)
+
+	for _, tc := range []struct {
+		name            string
+		folder          string
+		pluginJSON      []byte
+		failureFileName []string
+	}{
+		{
+			name:            "standalone-txt",
+			folder:          "standalone-txt",
+			pluginJSON:      pluginJSONWithExecutable,
+			failureFileName: []string{"standalone.txt"},
+		},
+		{
+			name:            "pid-txt",
+			folder:          "pid-txt",
+			pluginJSON:      pluginJSONWithExecutable,
+			failureFileName: []string{"pid.txt"},
+		},
+		{
+			name:            "all",
+			folder:          "all",
+			pluginJSON:      pluginJSONWithExecutable,
+			failureFileName: []string{"standalone.txt", "pid.txt"},
+		},
+		{
+			name:            "correct",
+			folder:          "correct",
+			pluginJSON:      pluginJSONWithExecutable,
+			failureFileName: []string{},
+		},
+		{
+			name:            "standalone-txt with no executable",
+			folder:          "standalone-txt",
+			pluginJSON:      pluginJSONEmpty,
+			failureFileName: []string{},
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
```

Comment:
I am not a fan of this kind of tests that test all conditions in a single loop. I rather have a loop for the possible failures and standalone tests for the correct case,